fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/ElectInstance.java

```diff
@@ -0,0 +1,137 @@
+package com.github.dedis.popstellar.model.objects;
+
+import androidx.annotation.NonNull;
+
+import com.github.dedis.popstellar.model.network.method.message.data.consensus.ConsensusElect;
+import com.github.dedis.popstellar.model.network.method.message.data.consensus.ConsensusElectAccept;
+import com.github.dedis.popstellar.model.network.method.message.data.consensus.ConsensusKey;
+import com.github.dedis.popstellar.model.objects.security.MessageID;
+import com.github.dedis.popstellar.model.objects.security.PublicKey;
+import com.github.dedis.popstellar.utility.security.Hash;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class holding information of an ConsensusElect message and its current states including the
+ * key/messageId of every node that have accepted this Elect with an ElectAccept.
+ */
+public final class ElectInstance {
+
+  private final MessageID messageId;
+  private final String channel;
+  private final PublicKey proposer;
+  private final ConsensusElect elect;
+  private final Set<PublicKey> nodes;
+  // map the public key of acceptors to the id of their ElectAccept message
+  private final Map<PublicKey, MessageID> acceptorToMessageId;
+
+  private State state;
+
+  public ElectInstance(
+      @NonNull MessageID messageID,
```

Comment:
It was because of the autocompletion that completed to messageID because the new type has a `D`.
I aggree that we should use only one of the two, but which one ?
If messageId, should I need to also change the class name to MessageId ?